memory_store.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials
+
+import (
+	"context"
+	"sync"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// MemoryStore is a store that keeps credentials in memory.
+type MemoryStore struct {
+	store sync.Map
+}
+
+// NewMemoryStore creates a new in-memory credentials store.
+func NewMemoryStore() *MemoryStore {
+	return &MemoryStore{}
+}
+
+// Get retrieves credentials from the store for the given server address.
+func (is *MemoryStore) Get(_ context.Context, serverAddress string) (auth.Credential, error) {
```

Comment:
nit:  `ms *MemoryStore`?  😀

Reply:
You may want to change the PR description as well.

Reply:
Thanks for your feedback!